cocos/core/data/decorators/property.ts

```diff
@@ -176,35 +176,21 @@ function genProperty (
             return;
         }
 
-        let defaultValue: any;
-        let isDefaultValueSpecified = false;
         if (descriptor) {
             // In case of Babel, if an initializer is given for class field.
             // That initializer is passed to `descriptor.initializer`.
             // babel
             if (descriptor.initializer) {
-                defaultValue = getDefaultFromInitializer(descriptor.initializer);
-                isDefaultValueSpecified = true;
+                propertyRecord.default = getDefaultFromInitializer(descriptor.initializer);
```

Comment:
"Whether a default value is specified" is decided by `hasOwnProperty()` instead of `undefined`.